eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs

```diff
@@ -225,6 +232,78 @@ protected MethodProvider BuildValidateResourceIdMethod()
             return new MethodProvider(signature, bodyStatements, this);
         }
 
+        private CSharpType GetPathParameterType(string parameterName)
+        {
+            var operation = InputOperationForRequestPath;
+
+            var operationParameter = operation.Parameters.First(p => p.Name.Equals(parameterName, StringComparison.OrdinalIgnoreCase));
+
+            var csharpType = ManagementClientGenerator.Instance.TypeFactory.CreateCSharpType(operationParameter.Type) ?? throw new InvalidOperationException($"Could not create C# type for parameter '{parameterName}' with type '{operationParameter.Type}' in operation '{operation.Name}'.");
+            return parameterName switch
+            {
+                "subscriptionId" when csharpType.Equals(typeof(Guid)) => typeof(string),
+                "location" when csharpType.Equals(typeof(string)) => typeof(AzureLocation),
+                // Cases will be added later
+                _ => csharpType
+            };
+        }
+
+        protected MethodProvider BuildCreateResourceIdentifierMethod()
+        {
+            var parameters = new List<ParameterProvider>();
+            var formatBuilder = new StringBuilder();
+            var first = true;
+            var refCount = 0;
+            foreach (var segment in _requestPath)
+            {
+                bool isConstant = RequestPath.IsSegmentConstant(segment);
+                if (!isConstant)
+                {
+                    var trimmed = RequestPath.TrimSegment(segment);
+                    var parameter = new ParameterProvider(trimmed, $"The {trimmed}", GetPathParameterType(trimmed));
+                    parameters.Add(parameter);
+                }
+
+                if (first)
+                {
+                    first = false;
+                    if (isConstant)
+                    {
+                        formatBuilder.Append('/');
+                    }
+                }
+                else
+                {
+                    formatBuilder.Append('/');
+                }
+
+                if (isConstant)
+                {
+                    formatBuilder.Append(segment);
+                }
+                else
+                {
+                    formatBuilder.Append($"{{{refCount++}}}");
```

Comment:
is it possible to have a UT for this case? curious how it looks in the final resource id.

Reply:
we already have a UT for it:
![image](https://github.com/user-attachments/assets/ad4a86a4-7d99-4f7e-a5a7-9da592dce579)